bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -130,35 +138,55 @@ public void updated(final Metadata oldMetadata, final Metadata newMetadata) {
         };
         itemRegistry.addRegistryChangeListener(this);
         metadataRegistry.addRegistryChangeListener(metadataChangeListener);
-        itemRegistry.getItems().forEach(this::createRootAccessories);
         initialiseRevision();
-        makeNewConfigurationRevision();
-        logger.info("Created {} HomeKit items in instance {}.", accessoryRegistry.getAllAccessories().size(), instance);
+        boolean changed = false;
+        for (var i : itemRegistry.getItems()) {
+            String oldValue = knownAccessories.get(i.getName());
+            createRootAccessories(i);
+            if (accessoryChanged(i.getName(), oldValue)) {
+                logger.debug("Accessory {} changed:\n{}\n{}", i.getName(), oldValue, knownAccessories.get(i.getName()));
+                changed = true;
+            }
+        }
+        // order of this conditional is important - checkMissingAccessories has side effects that need to always happen
+        if (checkMissingAccessories() || changed) {
+            makeNewConfigurationRevision();
+        } else {
+            logger.info("Created {} HomeKit items in instance {} (no change from prior configuration).",
+                    accessoryRegistry.getAllAccessories().size(), instance);
+            checkForDummyAccessories();
```

Comment:
we need to execute check only if useDummy is true?
i..e
if (settings.useDummyAccessories) checkForDummyAccessories();